[TypeScript-Angular] Path URI Encoding Update

defmonk0 opened the pull request:
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/swagger-api/swagger-codegen/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the correct branch: `3.0.0` branch for changes related to OpenAPI spec 3.0. Default: `master`.
- [x] Copied the [technical committee](https://github.com/swagger-api/swagger-codegen/#swagger-codegen-technical-committee) to review the pull request if your PR is targeting a particular programming langauge.

### Description of the PR

This update replaces the code for updating paths for URI encoding. Previously, this code was not converting from snake_case to camelCase, causing the variables embedded in the path to not match the parameter names they were passed to the function as.

The new method involves finding matches of items in curly-braces and iterating through them. Each match is then searched through for underscores. If one is found, it is removed and the following letter is capitalized.

The more complicated replacement is necessary to guarantee we only change the format of the variables inside the template braces. Otherwise, it might replace parts of the url/path itself, rather than just the variable names.

This should fix #6762.

@TiFu @taxpon @sebastianhaas @kenisteward @Vrolijkx

macjohnny commented on modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/TypeScriptAngularClientCodegen.java:
```diff
@@ -247,8 +250,55 @@ public void postProcessParameter(CodegenParameter parameter) {
+            // This should be anything that was contained inside curly-braces.
+            while(pathMatcher.find()) {
+                // Get a substring of our match.
+                String templateSection = op.path.substring(pathMatcher.start(), pathMatcher.end());
```
I would choose a simpler implementation like e.g. https://stackoverflow.com/a/34229320

defmonk0 replied:
Hmm. Looking at it now, I agree. It's been a long while since I've coded in Java, so after seeing the original code used regex, I just did the same in case it needed to be more complicated later. That said, it's definitely not the most efficient. 

I'll mess with this is a bit, I should be able to do this with much simpler string manipulation, crawling through 1 character at a time. I should only need one iteration through the string.

defmonk0 commented:
Rebased on master, and re-pushed the branch with the extra commit.

I simplified the implementation. Much cleaner, and more efficient too. Since using split would have still required avoiding underscores outside of the template braces, I figured this iteration method was simpler in the long run. No regex or secondary iteration required.

defmonk0 commented:
I rebased on master again, so that the tests would pass (since the previous commit this was based on was not). @macjohnny thoughts?

wing328 commented:
@defmonk0 thanks for the PR with detailed comments in the code, which looks good to me.
